sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/interfaces.scala

```diff
@@ -169,6 +168,9 @@ abstract class AggregateFunction2
 
   override protected def genCode(ctx: CodeGenContext, ev: GeneratedExpressionCode): String =
     throw new UnsupportedOperationException(s"Cannot evaluate expression: $this")
+
+  // Some aggregate function like Hive UDAF doesn't support partial aggregation, should override it.
+  def supportPartial: Boolean = true
```

Comment:
How about we name it as `supportsPartial`? Also, for the doc, let's make it as Scala doc, e.g.
```
/** Indicates if this function supports partial aggregation. */
```